doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+16           WARNING    PSU_FAULT                      2021-05-20T02:16:42.611Z    :- /psu/2 has experienced a fault 
+
+sonic# show alarm summary
+Alarm summary
+---------------------------------
+Total:                         3 
+Critical:                      0 
+Major:                         0 
+Minor:                         0 
+Warning:                       3 
+Acnowledged:                   2 
+----------------------------------
+```
+
+### 3.3.3 REST API Support
+
+sonic REST links:
+*  /restconf/data/sonic-event:sonic-event/EVENT/EVENT_LIST
+*  /restconf/data/sonic-event:sonic-event/EVENT_STATS/EVENT_STATS_LIST
+*  /restconf/data/sonic-alarm:sonic-alarm/ALARM/ALARM_LIST
+*  /restconf/data/sonic-alarm:sonic-alarm/ALARM_STATS/ALARM_STATS_LIST
+*  /restconf/operations/sonic-evprofile:get-evprofile
+*  /restconf/operations/sonic-evprofile:set-evprofile
+*  /restconf/operations/sonic-alarm:acknowledge-alarms
+*  /restconf/operations/sonic-alarm:unacknowledge-alarms
+
+openconfig REST links:
+*  /restconf/data/openconfig-system:system/openconfig-events:events
+*  /restconf/data/openconfig-system:system/openconfig-events:event-stats
+*  /restconf/data/openconfig-system:system/alarms
+*  /restconf/data/openconfig-system:system/openconfig-alarms-ext:alarm-stats
+
+# 4 Flow Diagrams
+![Sequence Diagram](event-alarm-framework-seqdiag.png)
+
+# 5 Warm Boot Support
+## 5.1 Application warm boot
+Applications confirming to the warm boot, should have stored their state and compare current values against previous values.
+Such compliant application also "remembers" that it raised an event before for a specific condition.
+They would 
+*  not raise alarms/events for the same condition that it raised pre warm boot
+*  clear those alarms once current state of a particular condition is recovered (by comparing against the stored state). 
+
+## 5.2 eventd warm boot
+Records from applications are stored in a table, called EVENTPUBSUB.
+Records that are being written will be queued when the consumer (eventd) is down.
+
+During normal operation, eventd reads, processes whenever a new record is added to the table.
+
+When eventd is restarted, events and alarms raised by applications will be waiting in a queue while eventd is coming up.
+When eventd eventually comes back up, it reads those records in the queue.
+
+# 6 Scalability
+In this feature, scalability applies to Event History Table (EVENT). As it is persistent and it records every event generated on the system, to protect
+against it growing indefinitely, user can limit its size through a manifest file.
+By default, the size of Event History Table is set to 40k events or events for 30 days - after which, older records are discarded to make way for new records.
+
+# 7 Showtech support
+The techsupport bundle is upgraded to include output of "show event recent 60min” and “show alarm all”.
```

Comment:
The show tech support can also be executed with the option --since.
I believe you should refer to that option in the design and if it is used make sure it is included in the option of the show event recent. Maybe it is even recommended to use the same format of the --since option as it is also taken for logs.

Reply:
This can be considered as an enhancement that can be brought in later to the commands.